src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -108,7 +109,16 @@ public void run(ActionContext context) throws Exception {
     LOG.debug("The BigQuery SQL is {}", config.getSql());
 
     // Wait for the query to complete
-    queryJob.waitFor();
+    try {
+      queryJob.waitFor();
+    } catch (BigQueryException e) {
+      // Minor bug with BigQuery java client causing it to prematurely throw an exception
```

Comment:
is this a known issue in the BQ client (is there a github issue or some other type of bug tracking it)? It would be nice to know if it's fixed in a newer client version, so that we can remove this handling if we ever upgrade the library past that version.

Reply:
I tested this on the newest Maven package version (116.6) and it was still not fixed. I plan on opening an issue with the BigQuery team about it.